ote/src/cljs/ote/app/routes.cljs

```diff
@@ -149,33 +149,34 @@
                      ;; Remove potentially sensitive arguments from analytics script reporting
                      win-location (if-let [opt-out-page (#{:register :reset-password :confirm-email} (:page navigation-data))]
                                     (str "/" (name opt-out-page))
-                                    win-location)]
-
-                 (if (or (requires-authentication? app)
-                         (requires-transit-authority? app)
-                         (requires-admin? app))
+                                    win-location)
+                     not-authorized? (or (requires-authentication? app)
+                                       (requires-transit-authority? app)
+                                       (requires-admin? app))]
+                 (if not-authorized?
                    (do (navigate! :login)
                        (assoc orig-app
                          :login {:show? true
                                  :navigate-to navigation-data}))
 
                    ;; Send startup events (if any) immediately after returning from this swap
-                   (when (or event-leave event-to)
-
-                     ; SPA page changes must be pushed to analytics script because url route might not change.
-                     ;; Check tracker script in case script loading failed. A browser extension or other issue may block it.
-                     (when (exists? js/_paq)
-                       (.push js/_paq (clj->js ["setCustomUrl", win-location]))
-                       ;; trackPageView signals to matomo piwik js api a single page visit.
-                       (.push js/_paq (clj->js ["trackPageView"])))
-
-                     (.setTimeout
-                       js/window
-                       (fn []
-                         (send-startup-events (vec (concat event-leave event-to))))
-                       0)
-                     app))
-                 app))))))
+                   (if (or event-leave event-to)
+                     (do
```

Comment:
Is "do" really needed here?

Reply:
Yes, since we want to do more than one thing in the positive outcome of the if.